src/core/dense_set.cc

```diff
@@ -180,21 +180,34 @@ void* DenseSet::PopDataFront(DenseSet::ChainVectorIterator it) {
 }
 
 void DenseSet::ClearInternal() {
+  constexpr unsigned kArrLen = 32;
+  ClearItem arr[kArrLen];
+  unsigned len = 0;
+
   for (auto it = entries_.begin(); it != entries_.end(); ++it) {
-    while (!it->IsEmpty()) {
-      bool has_ttl = it->HasTtl();
-      bool is_displ = it->IsDisplaced();
-      void* obj = PopDataFront(it);
-      int32_t delta = int32_t(BucketId(obj, 0)) - int32_t(it - entries_.begin());
-      if (is_displ) {
-        DCHECK(delta < 2 || delta > -2);
-      } else {
-        DCHECK_EQ(delta, 0);
-      }
-      ObjDelete(obj, has_ttl);
+    if (it->IsEmpty())
+      continue;
+
+    DensePtr ptr = *it;
+    auto& dest = arr[len++];
+    dest.has_ttl = ptr.HasTtl();
+
+    PREFETCH_READ(ptr.Raw());
+    if (ptr.IsObject()) {
+      dest.obj = ptr.Raw();
+      dest.ptr.Reset();
+    } else {
+      dest.ptr = ptr;
+      dest.obj = nullptr;
+    }
```

Comment:
You merge very similar code in https://github.com/dragonflydb/dragonfly/pull/3862/files#diff-5685b15a1d75a2c80b43908203ee74ab673f9a748f1dce35ad1f018bb656eb6aR342 maybe worth extracting it into a function with an optional callback ?